source/constructs/api/data_source/service.py

```diff
@@ -1193,41 +1212,221 @@ def before_delete_rds_connection(account: str, region: str, instance: str):
                                MessageEnum.DISCOVERY_JOB_CAN_NOT_DELETE_DATABASE.get_msg())
 
 
-def delete_rds_connection(account: str, region: str, instance: str):
+def delete_rds_connection(account: str, region: str, instance: str, delete_catalog_only=False):
     before_delete_rds_connection(account, region, instance)
     glue = __glue(account, region)
     rds_instance = crud.get_rds_instance_source(account, region, instance)
     err = []
     # 1/3 delete job database
     try:
-        delete_job_database(account_id=account, region=region, database_type='rds', database_name=instance)
+        if not delete_catalog_only:
+            delete_job_database(account_id=account, region=region, database_type='rds', database_name=instance)
     except Exception as e:
         err.append(str(e))
     # 2/3 delete catalog
     try:
         delete_catalog_by_database_region(database=instance, region=region, type='rds')
     except Exception as e:
         err.append(str(e))
-    # 3/3 delete source
-    try:
-        glue.delete_crawler(Name=rds_instance.glue_crawler)
-    except Exception as e:
-        err.append(e)
-    try:
-        glue.delete_database(Name=rds_instance.glue_database)
-    except Exception as e:
-        err.append(e)
-    try:
-        glue.delete_connection(ConnectionName=rds_instance.glue_connection)
-    except Exception as e:
-        err.append(e)
-    crud.delete_rds_connection(account, region, instance)
-    if err:
-        logger.error(traceback.format_exc())
-        # raise BizException(MessageEnum.SOURCE_RDS_CONNECTION_DELETE_ERROR.get_code(), f"Delete with error: {err}")
+
+    if not delete_catalog_only:
+        # 3/3 delete source
+        try:
+            glue.delete_crawler(Name=rds_instance.glue_crawler)
+        except Exception as e:
+            err.append(e)
+        try:
+            glue.delete_database(Name=rds_instance.glue_database)
+        except Exception as e:
+            err.append(e)
+        try:
+            glue.delete_connection(ConnectionName=rds_instance.glue_connection)
+        except Exception as e:
+            err.append(e)
+        crud.delete_rds_connection(account, region, instance)
+        if err:
+            logger.error(traceback.format_exc())
+            # raise BizException(MessageEnum.SOURCE_RDS_CONNECTION_DELETE_ERROR.get_code(), f"Delete with error: {err}")
 
     return []
 
+def hide_rds_connection(account: str, region: str, instance: str):
+    before_delete_rds_connection(account, region, instance)
+    crud.hide_rds_connection(account, region, instance)
+
+    return []
+
+def sync_glue_database_source(account: str, region: str, database_name: str):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 78 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.